### PR TITLE
Replacement of legacy devices with `default.qubit`

### PR DESCRIPTION
**Pull request by Shiro-Raven**
**Summary:**
The legacy qubit devices `default.qubit.{tf,torch,jax,autograd,legacy}` are now deprecated and need to be replaced with `default.qubit`.

[[sc-70260](https://app.shortcut.com/xanaduai/story/70260)]

### CODE REVIEW

File: demonstrations/learning2learn.py [hunk 1]
```diff
@@ -373,9 +372,7 @@ def recurrent_loop(graph_cost, n_layers=1, intermediate_steps=False):
     # We perform five consecutive calls to 'rnn_iteration', thus creating the
     # recurrent loop. More iterations lead to better results, at the cost of
     # more computationally intensive simulations.
-    out0 = rnn_iteration(
-        [initial_cost, initial_params, initial_h, initial_c], graph_cost
-    )
+    out0 = rnn_iteration([initial_cost, initial_params, initial_h, initial_c], graph_cost)
```
Comment (albi3ro) [inline demonstrations/learning2learn.py:375]: just line length formatting?

Comment (Shiro-Raven) [reply demonstrations/learning2learn.py:375]: Yup

**Review by albi3ro — COMMENTED** [review]
I'm double checking the output of the `learning2learn` demo by getting a crash. But it may be due to me not setting up Keras properly.  Think I've made mistakes in Keras versioning in the past. Stay tuned...

**Comment by Shiro-Raven** [comment]
> I'm double checking the output of the `learning2learn` demo by getting a crash. But it may be due to me not setting up Keras properly. Think I've made mistakes in Keras versioning in the past. Stay tuned...

Maybe this link is what you are looking for? 
https://qml-build-previews.pennylane.ai/pull_request_build_preview/1176/qml/demos/learning2learn/

**Review by PietropaoloFrisoni — APPROVED** [review]
Thank you @Shiro-Raven , looks good to me!

@albi3ro the `learning2learn.py` demo fails with the latest version of `tensorflow` and `keras`, but on my laptop works with `tensorflow-2.14.1` and `keras-2.14.0` (the same used in the `poetry.lock` file)

File: demonstrations/ahs_aquila.py [hunk 2]
```diff
@@ -512,7 +512,7 @@ def gaussian_fn(p, t):
 params = [amplitude_params]
 ts = [0.0, 1.75]
 
-default_qubit = qml.device("default.qubit.jax", wires=3, shots=1000)
+default_qubit = qml.device("default.qubit", wires=3, shots=1000)
```
Comment (albi3ro) [inline demonstrations/ahs_aquila.py:515]: Currently issues with the probability distribution not being normalized, but we can fix that independently afterward.

File: demonstrations/tutorial_jax_transformations.py [hunk 3]
```diff
@@ -253,8 +259,8 @@ def circuit(param):
 # To learn more about how JAX handles randomness, visit their
 # `documentation site. <https://jax.readthedocs.io/en/latest/jax.random.html>`__
 #
-# .. note:: 
-#     This example only applies if you are using ``jax.jit``. Otherwise, PennyLane 
+# .. note::
```
Comment (albi3ro) [inline demonstrations/tutorial_jax_transformations.py:262]: Trying to figure out what was meant by this statement. 

I'm tempted to say we just get rid of it, because I find it confusing.  What about this example only works with jax jit?

Maybe we can add a comment that seed can also accept any integer.

**Review by albi3ro — APPROVED** [review]
We should independently resolve the probability normalization issue, but I think we can consider that a different task.

For the note in `tutorial_jax_transformations.py`, I'll think about that note and come back to it. But for now, let's just resolve that conversation and move on.
